src/components/asset-panel/selector.jsx

```diff
@@ -0,0 +1,61 @@
+const React = require('react');
+
+const SpriteSelectorItem = require('../sprite-selector-item/sprite-selector-item.jsx');
+const Box = require('../box/box.jsx');
+const styles = require('./selector.css');
+
+const Selector = props => {
+    const {
+        items,
+        newText,
+        selectedItemIndex,
+        onDeleteClick,
+        onItemClick,
+        onNewClick
+    } = props;
+
+    return (
+        <Box className={styles.wrapper}>
+            <Box
+                className={styles.newItem}
+                onClick={onNewClick}
+            >
+                {newText}
+            </Box>
+            <Box className={styles.listArea}>
+                {items.map((item, index) => {
+                    const _onItemClick = () => onItemClick(item);
+                    const _onDeleteClick = e => {
+                        e.stopPropagation();
+                        onDeleteClick(item);
+                    };
```

Comment:
A couple of things about these handlers:
* Since these are behavioral, they should be owned by a container
* Function definitions in `render` should be avoided for performance (which is why the linter complains about arrow functions as props). If you can't convert the component to a class and use methods instead, a wrapper component would be another way to avoid this (this is partially the reason for the `SpriteSelectorItem` component itself).